test/save_assemblies_from_fastas_integration_test.py

```diff
@@ -576,4 +584,171 @@ def test_fail_mismatched_inputs(config, impl, context, scratch):
+        shutil.copy(data / file_name, tmp_dir)
+
+    params = {
+        'workspace_id': config['ws_id'],
+        'inputs': [
+            {'file': tmp_dir / 'GCA_002506415.1_ASM250641v1_genomic.fna.gz', 'assembly_name': 'GCA_002506415.1_ASM250641v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000007065.1_ASM706v1_genomic.fna.gz', 'assembly_name': 'GCF_000007065.1_ASM706v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000970165.1_ASM97016v1_genomic.fna.gz', 'assembly_name': 'GCF_000970165.1_ASM97016v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000970185.1_ASM97018v1_genomic.fna.gz', 'assembly_name': 'GCF_000970185.1_ASM97018v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000970205.1_ASM97020v1_genomic.fna.gz', 'assembly_name': 'GCF_000970205.1_ASM97020v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000970245.1_ASM97024v1_genomic.fna.gz', 'assembly_name': 'GCF_000970245.1_ASM97024v1_genomic.fna.gz'}
+        ]
+    }
+
+    object_version_pattern = re.compile(r'^[0-9]+\/1$')
+    results = impl.save_assemblies_from_fastas(context, params)[0]['results']
+    ws = _get_workspace(config)
+    for idx, res in enumerate(results):
+        # check the workspace is returning UPAs
+        assert res['filtered_input'] is None
+        assert object_version_pattern.match("/".join(res['upa'].split("/")[-2:]))
+
+        # check relevant object info fields
+        obj = ws.get_objects2({'objects': [{'ref': res['upa']}]})['data'][0]
+        info = obj['info']
+        assert info[1] == file_names[idx]
+        assert info[2].split('-')[0] == 'KBaseGenomeAnnotations.Assembly'
+        assert info[6] == config['ws_id']
+        assert info[10] == object_metas[idx]
+
+
+def test_invalid_thread_param(config, context, scratch):
+    tmp_dir = scratch / ("test_invalid_thread_param" + str(uuid.uuid4()))
+    os.makedirs(tmp_dir)
+    data = Path('data')
+    file_names = [
+        'GCA_002506415.1_ASM250641v1_genomic.fna.gz',
+        'GCF_000007065.1_ASM706v1_genomic.fna.gz'
+    ]
+
+    # copy 2 assembly files into the data dir
+    for file_name in file_names:
+        shutil.copy(data / file_name, tmp_dir)
+
+    params = {
+        'workspace_id': config['ws_id'],
+        'inputs': [
+            {'file': tmp_dir / 'GCA_002506415.1_ASM250641v1_genomic.fna.gz', 'assembly_name': 'GCA_002506415.1_ASM250641v1_genomic.fna.gz'},
+            {'file': tmp_dir / 'GCF_000007065.1_ASM706v1_genomic.fna.gz', 'assembly_name': 'GCF_000007065.1_ASM706v1_genomic.fna.gz'}
+        ]
+    }
+
+    var_name = "MAX_THREADS"
+    config['file_config']["KBASE_SECURE_CONFIG_PARAM_MAX_THREADS"] = "10e"
+    impl = AssemblyUtil(config['file_config'])
+    with raises(Exception) as got:
+        impl.save_assemblies_from_fastas(context, params)
+    assert_exception_correct(got.value, ValueError(f"Cannot evaluate the string {var_name}"))
+
+    config['file_config']["KBASE_SECURE_CONFIG_PARAM_MAX_THREADS"] = "-1"
```

Comment:
Won't this leave things in a failing state for future tests? If so copying the dict would fix that

Reply:
Updated!

Reply:
:+1: